redis/src/cluster_async/connections_logic.rs

```diff
@@ -429,10 +437,36 @@ async fn create_connection<C>(
 where
     C: ConnectionLike + Connect + Send + 'static,
 {
+    let retry_strategy = ExponentialBackoff {
+        initial_interval: std::time::Duration::from_millis(
+            params.exponential_backoff.factor as u64,
+        ),
+        multiplier: params.exponential_backoff.exponent_base as f64,
+        max_elapsed_time: None,
+        ..Default::default()
+    };
     let connection_timeout = params.connection_timeout;
     let response_timeout = params.response_timeout;
-    let info = get_connection_info(node, params)?;
-    C::connect(info, response_timeout, connection_timeout, socket_addr).await
+    let info = get_connection_info(node, params.clone())?;
+    let counter = std::sync::atomic::AtomicU32::new(params.exponential_backoff.number_of_retries);
```

Comment:
My only concern is that - if i'm not mistaking - when we try to refresh connections we actually block the entire client from new requests, since we lock the connection container with the write lock, right?
so what's the number_of_retries, for a single node? and what's the max waiting time? 
since refresh_connections can be called with multiple connection identifiers, we should see if we need to free the lock in some stages so the client won't get fully blocked (for example, retry the refresh_connections function, rather than per node)